### PR TITLE
Cleanup docs on users and roles

### PR DESCRIPTION
**Pull request by klizhentas**
- Updates roles to use v4
- Moves users to a separate guide
- Creates up to date reference for users and roles resources

### CODE REVIEW

File: docs/pages/setup/admin/users.mdx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+title: Local users
+description: Adding and deleting local users
+---
+
+This guide will cover Inviting and managing local user accounts.
```
Comment (r0mant) [inline docs/pages/setup/admin/users.mdx]: Is `Inviting` capitalized on purpose?

File: docs/pages/setup/reference/resources.mdx [hunk 2]
```diff
@@ -121,3 +121,265 @@ $ tctl rm users/admin
...
+      # User can access any region in us-west, e.g us-west-1, us-west-2
+      'region': 'us-west-*'
+      # regular expressions start with ^ and ending with $
+      # Teleport uses golang regexp syntax.
+      'cluster_name': '^us.*\.example\.com$'
+
+    # Functions transform variables.
+    database_users: ['{{email.local(external.email)}}']
+    database_labels:
+      'env': '{{regexp.replace(external.access["env"], "^(staging)$", "$1")}}'
```
Comment (r0mant) [inline docs/pages/setup/reference/resources.mdx:259]:
```suggestion
    db_users: ['{{email.local(external.email)}}']
    db_names: ['{{external.db_names}}']
    db_labels:
      'env': '{{regexp.replace(external.access["env"], "^(staging)$", "$1")}}'
```

**Review by r0mant — APPROVED** [review]
A typo in database related role fields, otherwise lgtm.

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**